doc/faq/readme.rst

```diff
@@ -139,3 +145,17 @@ The following lines are compatible with both lwIP versions:
     }
 
 Ref.  `#1923 <https://github.com/esp8266/Arduino/issues/1923>`__
+
+
+Why is there a board generator and what about it ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The board generator is a python script originally intended to ease the
+Arduino IDE's `boards.txt` configuration file about the multitude of
+available board, especially when common data have to be updated for all of
+them.
+
+This script is also used to manage uncommon options that will not find their
```

Comment:
that are currently not available in the IDE menu (we can't say for sure that they will never ever be in the IDE :p)